code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/BasicNotificationBuilderTest.kt

```diff
@@ -0,0 +1,218 @@
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.builders
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.BasicPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MOCKED_CHANNEL_ID
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MOCK_REMIND_LATER_DURATION
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MOCK_REMIND_LATER_TEXT
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MOCK_REMIND_LATER_TIME
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockAEPPushTemplateDataProvider
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.provideMockedBasicPushTemplateWithAllKeys
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.provideMockedBasicPushTemplateWithRequiredData
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.MapData
+import junit.framework.TestCase.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+import kotlin.test.assertNotNull
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class BasicNotificationBuilderTest {
+
+    @Mock
+    private lateinit var basicPushTemplate: BasicPushTemplate
+    private lateinit var context: Context
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var broadcastReceiverClass: Class<out BroadcastReceiver>
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+
+        context = RuntimeEnvironment.getApplication()
+        trackerActivityClass = DummyActivity::class.java
+        broadcastReceiverClass = DummyBroadcastReceiver::class.java
+    }
+
+    @Test
+    fun `construct should not return null`() {
+        val notificationBuilder = BasicNotificationBuilder.construct(context, basicPushTemplate, trackerActivityClass, broadcastReceiverClass)
+        assertNotNull(notificationBuilder)
+    }
+
+    @Test
+    fun `construct should return a NotificationCompat Builder`() {
+        val notificationBuilder = BasicNotificationBuilder.construct(context, basicPushTemplate, trackerActivityClass, broadcastReceiverClass)
+        assertEquals(NotificationCompat.Builder::class.java, notificationBuilder.javaClass)
+    }
+
```

Comment:
For these tests below can you also add assertions to verify that all the attributes are assigned as desired when a Notification is created from BasicTemplate?

Reply:
Added.